changelog/27790.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```

Comment:
While I agree that this fixes buggy behaviour, this is, strictly speaking, an API behaviour change -- so maybe this should be a `change` instead of an `improvement`?

Reply:
I agree with this. The current output is inconsistent but it is always possible that someone has become dependent on the existing behavior.